### PR TITLE
Switch InstallPluginActionTests to non-blocking SecureRandom seed generator

### LINKED ISSUE #102711: [CI] InstallPluginActionTests testOfficialPluginSnapshot failing
Zombie threads after the test class has run!

**Build scan:**
https://gradle-enterprise.elastic.co/s/22o6baukcol3k/tests/:distribution:tools:plugin-cli:test/org.elasticsearch.plugins.cli.InstallPluginActionTests/testOfficialPluginSnapshot%20%7Bp0=com.google.common.jimfs.JimfsFileSystem@3d49fd31%20p1=org.elasticsearch.plugins.cli.InstallPluginActionTests$1Parameter$$Lambda%2F0x0000000401326ae0@4016ccc1%7D

**Reproduction line:**
```
./gradlew ':distribution:tools:plugin-cli:test' --tests "org.elasticsearch.plugins.cli.InstallPluginActionTests" -Dtests.method="testOfficialPluginSnapshot {p0=com.google.common.jimfs.JimfsFileSystem@3d49fd31 p1=org.elasticsearch.plugins.cli.InstallPluginActionTests$1Parameter$$Lambda/0x0000000401326ae0@4016ccc1}" -Dtests.seed=731BAED44590999D -Dtests.locale=et -Dtests.timezone=America/Barbados -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testOfficialPluginSnapshot%20%7Bp0%3Dcom.google.common.jimfs.JimfsFileSystem@3d49fd31%20p1%3Dorg.elasticsearch.plugins.cli.InstallPluginActionTests$1Parameter$$Lambda/0x0000000401326ae0@4016ccc1%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.plugins.cli.InstallPluginActionTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium))))

**Failure excerpt:**
```
java.lang.Exception: Test abandoned because suite timeout was reached.

  at __randomizedtesting.SeedInfo.seed([731BAED44590999D]:0)

```

### PR DESCRIPTION
**Pull request by ldematte**
On Linux, the NativePRNG uses by default /dev/random as a source of randomness for seeds; according to comments on the [class itself](https://github.com/openjdk/jdk/blob/master/src/java.base/unix/classes/sun/security/provider/NativePRNG.java)

>     On some Unix platforms, /dev/random may block until enough entropy is
>     available, but that may negatively impact the perceived startup
>     on some versions of Linux entropy can be exhausted very quickly and could thus impact
>     startup time.

This PR makes SecureRandom use the NONBLOCKING approach by forcing to use `/dev/urandom`.
It may be less secure than using `/dev/random`, but this should be not too important in tests.

Closes https://github.com/elastic/elasticsearch/issues/102711

### CODE REVIEW

File: distribution/tools/plugin-cli/build.gradle [hunk 1]
```diff
@@ -38,6 +40,13 @@ tasks.named("dependencyLicenses").configure {
 tasks.named("test").configure {
   // TODO: find a way to add permissions for the tests in this module
   systemProperty 'tests.security.manager', 'false'
+  // These tests are "heavy" on the secure number generator. On Linux, the NativePRNG defaults to /dev/random for the seeds, and
+  // its entropy is quite limited, to the point that it's known to hang: https://bugs.openjdk.org/browse/JDK-6521844
+  // We force the seed to be initialized from /dev/urandom, which is less secure, but in case of unit tests is not important.
+  if (OS.current() == OS.LINUX) {
+    // The extra /./ is to work around https://bugs.openjdk.org/browse/JDK-4705093. See https://bugs.openjdk.org/browse/JDK-6202721.
+    systemProperty 'java.security.egd', 'file:/dev/./urandom'
```
Comment (thecoop) [inline distribution/tools/plugin-cli/build.gradle:48]: There's an extra dot?

**Review by thecoop — APPROVED** [review]

**Comment by ldematte** [comment]
@elasticmachine rerun elasticsearch-ci/docs

File: distribution/tools/plugin-cli/build.gradle [hunk 2]
```diff
@@ -38,6 +40,12 @@ tasks.named("dependencyLicenses").configure {
 tasks.named("test").configure {
   // TODO: find a way to add permissions for the tests in this module
   systemProperty 'tests.security.manager', 'false'
+  // These tests are "heavy" on the secure number generator. On Linux, the NativePRNG defaults to /dev/random for the seeds, and
+  // its entropy is quite limited, to the point that it's known to hang: https://bugs.openjdk.org/browse/JDK-6521844
+  // We force the seed to be initialized from /dev/urandom, which is less secure, but in case of unit tests is not important.
+  if (OS.current() == OS.LINUX) {
```
Comment (mark-vieira) [inline distribution/tools/plugin-cli/build.gradle:46]: I'm wondering if this is something we should set across the board for all tests.

Comment (ldematte) [reply distribution/tools/plugin-cli/build.gradle:46]: Could be worth it, if we observer slowdowns or timeouts. This test might have been an exception as it uses "more" SecureRandom (to check pluigin signatures).

Comment (mosche) [reply distribution/tools/plugin-cli/build.gradle:46]: I think that's a good idea to avoid potential bottlenecks, and also there's no harm doing so!

**Pull request merged**